cmd/bucket-metadata.go

```diff
@@ -66,7 +64,7 @@ func (lc *localBucketMetaState) UpdateBucketListener(args *SetBucketListenerPeer
 	}
 
 	// Update in-memory notification config.
-	return globalEventNotifier.SetBucketListenerConfig(args.Bucket, args.LCfg)
+	return objAPI.RefreshBucketListenerConfig(args.Bucket)
```

Comment:
Updating notification system is not object layer thing.  It should be done by S3 layer (HTTP handlers).  Having `RefreshBucketListenerConfig()` in `objAPI` is wrong.

Reply:
RefreshBucketListenerconfig() is a self contained call without making any RPC requests, having that in the object layer is not a problem @balamurugana . what you don't want is notification such as sending RPC messages in object layer which your PR is bringing in -  since notification configs are cached at the object layer it makes sense to have load+cache type calls such as "Refresh" at object layer.

What S3 layer does is `UpdateBucketListener`

Reply:
@harshavardhana 
> RefreshBucketListenerconfig() is a self contained call without making any RPC requests, having that in the object layer is not a problem

It may not be a problem, it is wrong by the design.  Updating notification system needs to be done post operation after object layer's put bucket notification call in S3 layer.

Reply:
UpdateListener() is called after the PutBucketNotification call in S3 layer to make RPC requests. internally if UpdateListener calls an `objAPI.Refresh*` to refresh the cache for configs from the backend - I don't see a design problem.